### PR TITLE
Add unit tests setup for standalone and webworker runtimes

### PR DESCRIPTION
**Pull request by natdurlik**
## Problem
Currently we need to test changes in the standalone and webworker functions manually using implementations of language servers.
## Solution
Added unit tests setup for both standalone and webworker. Allows to test features passed to servers as part of the initialization.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtimes/runtimes/util/testingUtils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                return target[property as keyof typeof target]
+            }
+            const method = sinon.stub()
+            ;(target as any)[property] = method
+            return method
+        },
+    })
+}
+
+export type Features = {
```
Comment (volodkevych) [inline runtimes/runtimes/util/testingUtils.ts:28]: Let's define it as part of server interface? We keep creating this type/interface in every new place.

File: runtimes/runtimes/standalone.test.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                })
+            })
+        })
+
+        describe('Telemetry', () => {
+            describe('emitMetric', () => {
+                it('should use LSP connection telemetry log event', () => {
+                    let metric: any
+                    features.telemetry.emitMetric(metric)
+                    sinon.assert.calledOnceWithExactly(stubConnection.telemetry.logEvent as sinon.SinonStub, metric)
```
Comment (volodkevych) [inline runtimes/runtimes/standalone.test.ts:70]: I do not see big value in tests that check function calling another function, to be honest. What kind of tests do you plan to implement with this setup in the future?

File: runtimes/runtimes/util/testingUtils.ts (hunk 1 above)
Comment (natdurlik) [reply runtimes/runtimes/util/testingUtils.ts]: Moved `Features` to `/server-interface/server.ts`.

File: runtimes/runtimes/standalone.test.ts (hunk 2 above)
Comment (natdurlik) [reply runtimes/runtimes/standalone.test.ts:71]: I just put it as an example. If there is some logic on the runtime side, then tests can be added like that.

**Review by volodkevych — APPROVED** [review]

File: runtimes/runtimes/util/testingUtils.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import sinon from 'sinon'
+
+export function createStubFromInterface<T>(): sinon.SinonStubbedInstance<T> & T {
```
[the remark this answers is not in the document]
Comment (natdurlik) [reply runtimes/runtimes/util/testingUtils.ts:3]: You are right, there is `stubInterface` in `ts-sinon` that could be used instead.

**Pull request merged**